yastd/dbus_api.md

```diff
@@ -0,0 +1,202 @@
+    Example:
+
+      ToInstall(["cs_CZ", "de_DE"]) -> () # only lang codes from AvailableLanguages is supported
+
+#### Properties (all read only):
+
+-  AvailableLanguages -> array(struct(string LangId, string LangLabel, dict(string, variant) details))
+    List of all available languages to install on target system.
+    Example:
+
+      AvailableLanguages -> [["cs_CZ", "Czech", {}]] # it is lang code, human readable lang name and dict for future extensions to provide more data
+
+-  MarkedForInstall -> array(string LangId)
+    List of languages to install. Same format as ToInstall
+
+#### Signals:
+
+-  PropertiesChanged ( only standard one from org.freedesktop.DBus.Properties interface )
+
+
+notes:
+
+identifiers: maybe LanguageTag https://www.rubydoc.info/github/yast/yast-packager/master/LanguageTag
+- move it to yast-yast2
+- link to the standard from yard
+- see https://tools.ietf.org/html/rfc4647 Matching of Language Tags
+- see https://lists.opensuse.org/archives/list/yast-devel@lists.opensuse.org/message/D52PSZ7TRID2RVM6CE6K2C2RUNNGOS6Z/
+
+## get the list of base products
+
+Iface: o.o.YaST.Installer1.Software
+
+#### methods:
+
+-   SelectProduct(string ProductId) -> void
+    Select product for installation.
+    TODO: do we need version or arch or any other parameter? should we add generic dict(sv) to it?
+    Example:
+
+      InstallProduct("SLES") -> () # only name from available BaseProducts is supported
+
+#### Properties (all read only):
+
+-  AvailableBaseProducts -> array(struct(string ProductId, string ProductLabel, dict(string, variant) details))
+    List of all available base product to install on target system.
+    Note: List is sorted according to defined display order
+    Example:
+
+      AvailableBaseProducts -> [["SLES", "SUSE Linux Enterprise Server", {}]] # it is product name, human readable name and dict for future extensions to provide more data
+
+-  SelectedBaseProduct -> string ProductId
+    Base product selected for installation. It is always defined.
+    Example:
+
+      SelectedBaseProduct -> "SLES" # only name from available BaseProducts is supported
+
+#### Signals:
+
+-  PropertiesChanged ( only standard one from org.freedesktop.DBus.Properties interface )
+
+
+## interact with the storage proposal
+
+### Iface: o.o.YaST.Installer1.Storage
+
+#### methods:
+
+-  MarkForUse(array(o.o.YaST.Installer1.StorageBlockDevice Device)) -> void
+    set objects for use of installation. it means erase content of that devices
+    example:
+
+      MarkForUse([disk1,disk2partition2]) -> ()
+
+-  MarkForShrinking(array(o.o.YaST.Installer1.StorageBlockDevice Device)) -> void
+    set objects to allow shrink of them. it means keep content and reduce its free space.
+    example:
+
+      MarkForShrink([disk1,disk2partition2]) -> ()
+
+#### Properties (all read only):
+
+-  Disks -> array(o.o.YaST.Installer1.Storage.Drive)  # an object\_path whose object implements this interface
```

Comment:
```suggestion
-  Drives -> array(o.o.YaST.Installer1.Storage.Drive)  # an object\_path whose object implements this interface
```

For consistency, the method name and the interface of the return value should match, like everywhere else in this API